src/configs/mod.rs

```diff
@@ -422,11 +424,31 @@ impl WireguardNetworkInfo {
         }
     }
 
+    fn unfold_flags(&self, info: &PeerInfo) -> Result<PeerInfo, String> {
```

Comment:
I dislike how you hacked "unfold_flags" into what should've been pluggable flag system. No go, until you make it through apply_to_peer.

Reply:
So I need to handle `use-template` flags only inside `apply_to_peer` and `apply_to_interface`?

What is pluggable flag system, I'm not sure I understand?

Reply:
> So I need to handle use-template flags only inside apply_to_peer and apply_to_interface?

yes

basically, by "pluggable" I mean that there's an interface to make new plugins exposing new flags. It is not expandable automatically rn, but by dropping logic all around the code we postpone this moment even further — so apply_to_peer and apply_to_interface exist, and are flat.

Reply:
@cab404 Is it OK that I build the graph of template dependencies (for all peers) every time `apply_to_interface`, `apply_to_peer`, and `apply_to_configuration` is called? 
My plan is to collect all dependent flags, and then apply all of them (recursion of depth of 1).